services/galley/src/Galley/API/Action.hs

```diff
@@ -362,20 +364,17 @@ instance IsConversationAction ConversationAccessData where
     -- 'PrivateAccessRole' is for self-conversations, 1:1 conversations and
     -- so on; users are not supposed to be able to make other conversations
     -- have 'PrivateAccessRole'
-    when
-      ( PrivateAccess `elem` cupAccess target
-          || PrivateAccessRole == cupAccessRole target
-      )
-      $ throw InvalidTargetAccess
+    when (PrivateAccess `elem` cupAccess target || Set.null (cupAccessRole target)) $
+      throw InvalidTargetAccess
     -- Team conversations incur another round of checks
     case convTeam conv of
       Just _ -> do
         -- Access mode change might result in members being removed from the
         -- conversation, so the user must have the necessary permission flag
         ensureActionAllowed RemoveConversationMember self
       Nothing ->
-        when (cupAccessRole target == TeamAccessRole) $
-          throw InvalidTargetAccess
+        when (Set.fromList [TeamMemberAccessRole] == cupAccessRole target) $ throw InvalidTargetAccess
```

Comment:
```suggestion
        when (TeamMemberAccessRole `Set.member` cupAccessRole target) $ throw InvalidTargetAccess
```

Reply:
I think the original logic is correct, I added a comment.